### PR TITLE
EC-276 Use swagger annotations for describing routes

### PR DESCRIPTION
**Pull request by jhnsmth**
To check it live:
- run the explorer
- got to https://petstore.swagger.io 
- explore http://127.0.0.1:9052/api-docs/swagger.json

### CODE REVIEW

**Comment by jhnsmth** [comment]
A couple of questions
- Should description be added for all model fields ?
- Should api responses be added for all routes ?

File: src/main/scala/org/encryfoundation/explorer/db/models/Input.scala [hunk 1]
```diff
@@ -1,6 +1,13 @@
 package org.encryfoundation.explorer.db.models
 
-case class Input(id: String, txId: String, proofs: String)
+import io.swagger.annotations.{ApiModel, ApiModelProperty}
+
+import scala.annotation.meta.field
+
+@ApiModel(value="Input", description="Transaction input")
+case class Input(id: String,
+                 @(ApiModelProperty @field)(name = "parentId") txId: String,
```
Comment (Bromel777) [inline src/main/scala/org/encryfoundation/explorer/db/models/Input.scala]: remove parentId

File: src/main/scala/org/encryfoundation/explorer/http/api/routes/SwaggerRoute.scala [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import akka.http.scaladsl.server.Route
+import com.github.swagger.akka.SwaggerHttpService
+import com.github.swagger.akka.model.Info
+import io.swagger.models.auth.BasicAuthDefinition
+import org.encryfoundation.explorer.db.models._
+import org.encryfoundation.explorer.settings.RESTApiSettings
+
+case class SwaggerRoute(settings: RESTApiSettings)(implicit val context: ActorRefFactory) extends SwaggerHttpService with ApiRoute {
+
+  override val apiClasses = Set(classOf[HistoryRoute], classOf[TransactionsApiRoute])
```
Comment (oskin1) [inline src/main/scala/org/encryfoundation/explorer/http/api/routes/SwaggerRoute.scala]: Please add type annotation to each `val`.

File: src/main/scala/org/encryfoundation/explorer/http/api/routes/SwaggerRoute.scala [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import com.github.swagger.akka.model.Info
+import io.swagger.models.ExternalDocs
+import io.swagger.models.auth.{BasicAuthDefinition, SecuritySchemeDefinition}
+import org.encryfoundation.explorer.db.models._
+import org.encryfoundation.explorer.settings.RESTApiSettings
+
+case class SwaggerRoute(settings: RESTApiSettings)(implicit val context: ActorRefFactory) extends SwaggerHttpService with ApiRoute {
+
+  override val apiClasses: Set[Class[_]] = Set(classOf[HistoryRoute], classOf[TransactionsApiRoute])
+  override val host: String = "localhost:9052"
```
Comment (kapinuss) [inline src/main/scala/org/encryfoundation/explorer/http/api/routes/SwaggerRoute.scala]: it should be in config

Comment (jhnsmth) [reply src/main/scala/org/encryfoundation/explorer/http/api/routes/SwaggerRoute.scala]: done

**Pull request merged**